crates/iota-node/src/lib.rs

```diff
@@ -833,6 +843,34 @@ impl IotaNode {
         let iota_node_metrics =
             Arc::new(IotaNodeMetrics::new(&registry_service.default_registry()));
 
+        // --- Create shared checkpoint broadcast channel and buffer for gRPC and
+        // checkpoint logic ---
+        let (grpc_checkpoint_summary_tx, grpc_checkpoint_data_tx) = if let Some(grpc_api_address) =
+            config.grpc_api_address
+        {
+            let (summary_tx, _) = tokio::sync::broadcast::channel(CHECKPOINT_BROADCAST_BUFFER_SIZE);
```

Comment:
Those channel sizes should be part of the `iota_grpc_api::Config`

Reply:
Agreed. Fixed at [eda61fc](https://github.com/iotaledger/iota/pull/7511/commits/eda61fcd0ff85884d947ab1822faeead8df89a17).